Sources/DittoPresenceDegradation/PresenceDegradationVM.swift

```diff
@@ -0,0 +1,122 @@
+                
+                self.localPeer = Peer(
+                    name: graph.localPeer.deviceName,
+                    transportInfo: localPeerTransportInfo,
+                    connected: true,
+                    lastSeen: Int(seenAt.timeIntervalSince1970),
+                    key: self.hashPeerKeyUseCase(graph.localPeer.peerKey)
+                )
+                
+                for peer in graph.remotePeers {
+                    let hashedPeerKey = self.hashPeerKeyUseCase(peer.peerKey)
+                    if var remotePeers = self.remotePeers {
+                        remotePeers[hashedPeerKey] = Peer(
+                            name: peer.deviceName,
+                            transportInfo: self.resolveTransportInfo(peer: peer),
+                            connected: true,
+                            lastSeen: Int(seenAt.timeIntervalSince1970),
+                            key: hashedPeerKey
+                        )
+                        self.remotePeers = remotePeers
+                    } else {
+                        self.remotePeers = [hashedPeerKey: Peer(
+                            name: peer.deviceName,
+                            transportInfo: self.resolveTransportInfo(peer: peer),
+                            connected: true,
+                            lastSeen: Int(seenAt.timeIntervalSince1970),
+                            key: hashedPeerKey
+                        )]
+                    }
+                }
+                
+                if let peers = self.remotePeers?.values {
+                    for peer in peers {                    
+                        if !graph.remotePeers.contains(where: { self.hashPeerKeyUseCase($0.peerKey) == peer.key}) {
+                            self.remotePeers?[peer.key]?.connected = false
+                        }
+                    }
+                }
+            }
+        }
+        
+        self.updateSettings()
+    }
+
+    func resolveTransportInfo(peer: DittoPeer) -> PeerTransportInfo {
+        let lanSet: Set<DittoConnectionType> = [.accessPoint, .webSocket]
+
+        let connections = peer.connections.map{ $0.type }
+        let bluetoothConnections = connections.filter { $0 == DittoConnectionType.bluetooth }.count
+        let lanConnections = connections.filter { lanSet.contains($0) }.count
+        let p2pConnections = connections.filter { $0 == DittoConnectionType.p2pWiFi }.count
+        let cloudConnections = peer.isConnectedToDittoCloud ? 1 : 0
+        
+        return PeerTransportInfo(
+            bluetoothConnections: bluetoothConnections,
+            lanConnections: lanConnections,
+            p2pConnections: p2pConnections,
+            cloudConnections: cloudConnections
+        )
+    }
+    
+    func hashPeerKeyUseCase(_ data: Data) -> String {
+        let hash = Insecure.MD5.hash(data: data)
+        return hash.map { String(format: "%02hhx", $0) }.joined()
+    }
```

Comment:
I think this needs to be the URL safe base 64 encoding with "pk" stuck on the front to align to other Peer Key usage?